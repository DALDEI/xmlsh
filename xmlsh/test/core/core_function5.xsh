# Test of nesting and newlines in function args

function add()
{
  return <[ xs:decimal($_1) + xs:decimal($_2) ]> ;
}

function concat()
{
  [ $# -lt 2 ] && return $1 
  local _1=$1
  shift
  local _n=concat($*)
  return "$_1$_n"
}

echo concat( a b )

echo add( 1
2)

echo add( 
1
2
)

echo concat(  
  add(1 
  2)
add( 3 
 4) concat( add(
 3 1) concat( foo bar) 
 )
 )


x=concat(  
  add(1 
  2)
add( 3 
 4) concat( add(
 3 1) concat( foo bar) 
 )
 )

echo $x

function l() {
local _x=concat(  
  add(1 2)
add( 3 4) concat( add( 3 1) concat( 
foo bar) )
)
 return $_x
}

echo l()
  
