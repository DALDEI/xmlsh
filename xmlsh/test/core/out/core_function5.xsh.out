ab
3
3
374foobar
374foobar
374foobar
